frontend/plan/components/selector/Tag.tsx

```diff
@@ -1,24 +1,49 @@
 import React, { PropsWithChildren } from "react";
+import styled from "styled-components";
 
 interface TagProps {
     onClick?: () => void;
     isAdder?: boolean;
 }
 
+const TagContainer = styled.span`
```

Comment:
```suggestion
const TagContainer = styled.span<{isAdder: boolean}>`
```

You can specify the props interface in the definition so the lambdas further down can infer the props. Also it'll help with inference when you use the container. I might have missed this in other places, but you can do it everywhere you have custom props in a styled component.